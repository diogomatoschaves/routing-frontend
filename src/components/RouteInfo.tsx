import React from 'react'
import { Box, StyledSegment } from '../styledComponents'
import styled from 'styled-components'
import { Statistic, Header } from 'semantic-ui-react'
import BackgroundIcon from './BackgroundIcon'
import { NORMAL_INPUT, PROFILE_BACKGROUND } from '../utils/colours'
import { computeDuration, computeDistance } from '../utils/functions'


interface Props {
  route: {
    duration: number,
    distance: number,
  },
  top: string,
  right: string,
  title: string,
  subTitle?: string,
  textColor: string
  iconColor: string
  statsColor: string
}

const StyledHeader = styled(Header)`
  &.ui.header {
    color: ${props => props.overridecolor ? props.overridecolor : 'black'};
    margin: ${props => props.noMargin ? 0 : props.noMarginBottom ? '10px 0 0 0' : '10px 0'};
  }
`

const StyledStatistic = styled(Statistic)`
  &.ui.statistic > div{
    color: ${props => props.overridecolor ? props.overridecolor : 'black'};
  }
`

const diameter = 50

const RouteInfo: any = ({ right, top, route, title, textColor, iconColor, statsColor, subTitle } : Props) => {

  const [durationValue, durationLabel] = computeDuration(route.duration)
  const [distanceValue, distanceLabel] = computeDistance(route.distance)

  return (
    <StyledSegment
      position="absolute"
      right={right}
      top={top}
      zindex={1000}
      width="200px"
    >
      <Box direction="column" justify="space-between">
        <StyledHeader noMarginBottom overridecolor={textColor}>{title}</StyledHeader>
        {subTitle && <StyledHeader size="small" overridecolor={iconColor}>{subTitle}</StyledHeader>}
        <Box direction="row" justify="space-around" width="100%" padding="5px 0 5px 0">
          <Box direction="row" width="40%" justify="center">
            <BackgroundIcon 
              diameter={diameter}
              color={PROFILE_BACKGROUND}
              iconColor={iconColor}
              circle={false}
              iconName={'time'}
              margin={"0 7px 0 0"}
            />
          </Box>
          <Box direction="row" width="60%" justify="center">
            <StyledStatistic overridecolor={statsColor} size={"small"} style={{ margin: 0 }}>
              <Statistic.Value>
                {durationValue}
              </Statistic.Value>
              <Statistic.Label>
                {durationLabel}
              </Statistic.Label>
            </StyledStatistic>
          </Box>
        </Box>
        <Box direction="row" justify="space-around" width="100%" padding="5px 0 5px 0">
          <Box direction="row" width="40%" justify="center">
            <BackgroundIcon 
              diameter={diameter}
              color={PROFILE_BACKGROUND}
              iconColor={iconColor}
              circle={false}
              iconName={'road'}
              margin={"0 7px 0 0"}
            />
          </Box>
          <Box direction="row" width="60%" justify="center">
          <StyledStatistic overridecolor={statsColor} size={'small'} style={{ margin: 0 }}>
              <Statistic.Value>
                {distanceValue}
              </Statistic.Value>
              <Statistic.Label>
                {distanceLabel}
              </Statistic.Label>
            </StyledStatistic>
          </Box>
        </Box>
      </Box>
    </StyledSegment>
  )
}

export default RouteInfo
