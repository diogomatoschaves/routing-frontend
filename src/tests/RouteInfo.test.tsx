import React from 'react'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import RouteInfo from '../components/RouteInfo';

const mockRoute = {
  duration: null,
  distance: null,
  routePath: null
}

const mockColor = '#000000'

it('expect to render ProfileToggler component', () => {

  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <RouteInfo 
        statsColor={mockColor}
        textColor={mockColor}
        iconColor={mockColor}
        title={'Routing Service'}
        subTitle={'With Traffic'}
        route={mockRoute}
        top={'1px'}
        right={'1px'}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
