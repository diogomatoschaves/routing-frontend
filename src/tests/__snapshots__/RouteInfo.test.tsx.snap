// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ProfileToggler component 1`] = `
<div
  className="ui segment sc-ifAKCX bRACfR"
  noBoxShadow={false}
  padding="15px 25px"
  width="100%"
>
  <div
    className="sc-bdVaJa fKJBrF"
    direction="column"
  >
    <div
      className="ui header sc-gqjmRU eAnDBN"
      nomarginbottom={1}
      overridecolor="#000000"
    >
      Routing Service
    </div>
    <div
      className="ui small header sc-gqjmRU jlzsIH"
      overridecolor="#000000"
    >
      With Traffic
    </div>
    <div
      className="sc-bdVaJa jzIkVo"
      direction="row"
      width="100%"
    >
      <div
        className="sc-bdVaJa ehaESb"
        direction="row"
        width="40%"
      >
        <div
          className="sc-gZMcBi gyuikZ"
        >
          <div
            className="sc-bwzfXH lYGau"
            color="#e1e1e1"
          >
            <div
              className="sc-bdVaJa hVRN"
              height="100%"
            >
              <i
                aria-hidden="true"
                className="time icon sc-htpNat kOxnLf"
                fontSize="25px"
                height="20px"
                onClick={[Function]}
                overridecolor="#000000"
                position="absolute"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-bdVaJa jEnIfZ"
        direction="row"
        width="60%"
      >
        <div
          className="ui tiny statistic sc-VigVT gGdolP"
          overridecolor="#000000"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          <div
            className="value"
          >
            0
          </div>
          <div
            className="label"
          >
            seconds
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa jzIkVo"
      direction="row"
      width="100%"
    >
      <div
        className="sc-bdVaJa ehaESb"
        direction="row"
        width="40%"
      >
        <div
          className="sc-gZMcBi gyuikZ"
        >
          <div
            className="sc-bwzfXH lYGau"
            color="#e1e1e1"
          >
            <div
              className="sc-bdVaJa hVRN"
              height="100%"
            >
              <i
                aria-hidden="true"
                className="road icon sc-htpNat kOxnLf"
                fontSize="25px"
                height="20px"
                onClick={[Function]}
                overridecolor="#000000"
                position="absolute"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-bdVaJa jEnIfZ"
        direction="row"
        width="60%"
      >
        <div
          className="ui tiny statistic sc-VigVT gGdolP"
          overridecolor="#000000"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          <div
            className="value"
          >
            0
          </div>
          <div
            className="label"
          >
            meters
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
